ksql-engine/src/main/java/io/confluent/ksql/util/KafkaTopicClientImpl.java

```diff
@@ -295,30 +304,47 @@ private void validateTopicProperties(String topic, int numPartitions, short repl
     );
   }
 
-  private static class RetryHelper<T> {
-    T executeWithRetries(Supplier<KafkaFuture<T>> supplier) throws  InterruptedException,
-                                                                    ExecutionException {
-      int retries = 0;
-      Exception lastException = null;
-      while (retries < NUM_RETRIES) {
-        try {
-          if (retries != 0) {
-            Thread.sleep(RETRY_BACKOFF_MS);
-          }
-          return supplier.get().get();
-        } catch (ExecutionException e) {
-          if (e.getCause() instanceof  RetriableException) {
-            retries++;
-            log.info("Retrying admin request due to retriable exception. Retry no: "
-                     + retries, e);
-            lastException = e;
-          } else {
-            throw e;
-          }
+  private Map<String, String> getTopicConfig(final String topicName,
+                                             final boolean includeDefaults) {
+    final ConfigResource resource = new ConfigResource(ConfigResource.Type.TOPIC, topicName);
+    final List<ConfigResource> request = Collections.singletonList(resource);
+
+    try {
+      final Config config = executeWithRetries(() -> adminClient.describeConfigs(request).all())
+          .get(resource);
+
+      return config.entries().stream()
+          .filter(e -> includeDefaults
+                       || e.source().equals(ConfigEntry.ConfigSource.DYNAMIC_TOPIC_CONFIG))
+          .collect(Collectors.toMap(ConfigEntry::name, ConfigEntry::value));
+    } catch (InterruptedException | ExecutionException e) {
+      throw new KafkaResponseGetFailedException(
+          "Failed to get config for Kafka Topic " + topicName, e);
+    }
+  }
+
+  private static <T> T executeWithRetries(
+      final Supplier<KafkaFuture<T>> supplier) throws InterruptedException,
+                                                      ExecutionException {
+    int retries = 0;
+    Exception lastException = null;
+    while (retries < NUM_RETRIES) {
```

Comment:
not a massive fan of this style of control logic - but hey - it works ;)

Reply:
It's not code I added. I only removed the redundant surrounding class.

I have no issue with retry strategies, but IMHO the strategy, or at least some of its params, should be configurable.

Reply:
@bluemonk3y any suggestions on how to improve it?

Reply:
@big-andy-coates not sure I agree that the configuration retry parameters here (the number of retries and the retry backoff) need to be exposed as a top level config. This is the kind of in-the-bowels detail which nobody should have to touch in normal conditions. This is especially true if we update the code to be more conservative and do things like exponential backoffs, etc.. Once this is more robust, we should only run out of failures if there is a hard downtime on the cluster, in which case tuning any parameters will not help.

Reply:
Hey @apurvam, it's not a big thing. The whole 'more dials' vs 'less dials' to tune is always an interesting topic to discuss. 

I fear we're getting off topic ;). This retry logic is not new code and so should really be outside the scope of this PR anyway.